src/providers/postgres/qgspostgresprojectstorage.cpp

```diff
@@ -0,0 +1,188 @@
+  {
+    // try to create projects table
+    QString sql = QStringLiteral( "CREATE TABLE %1.qgis_projects(name TEXT PRIMARY KEY, metadata JSONB, content BYTEA)" ).arg( projectUri.schemaName );
+    QgsPostgresResult res( conn->PQexec( sql ) );
+    if ( res.PQresultStatus() != PGRES_COMMAND_OK )
+    {
+      QString errCause = QObject::tr( "Unable to save project. It's not possible to create the destination table on the database. Maybe this is due to table permissions (user=%1). Please contact your database admin" ).arg( projectUri.connInfo.username() );
+      context.pushMessage( errCause, Qgis::Critical );
+      QgsPostgresConnPool::instance()->releaseConnection( conn );
+      return false;
+    }
+  }
+
+  // read from device and write to the table
+  QByteArray content = device->readAll();
+
+  QString metadata = "{ \"last_modified\": 123 }";  // TODO: real metadata
+
+  // TODO: would be useful to have QByteArray version of PQexec() to avoid bytearray -> string -> bytearray conversion
+  QString sql( "INSERT INTO %1.qgis_projects VALUES (%2, %3, E'\\\\x" );
+  sql = sql.arg( QgsPostgresConn::quotedIdentifier( projectUri.schemaName ),
+                 QgsPostgresConn::quotedValue( projectUri.projectName ),
+                 QgsPostgresConn::quotedValue( metadata ) );
+  sql += QString::fromAscii( content.toHex() );
+  sql += "') ON CONFLICT (name) DO UPDATE SET content = EXCLUDED.content, metadata = EXCLUDED.metadata;";
+
+  QgsPostgresResult res( conn->PQexec( sql ) );
+  bool ok = res.PQresultStatus() == PGRES_COMMAND_OK;
+
+  QgsPostgresConnPool::instance()->releaseConnection( conn );
+
+  return ok;
+}
+
+
+bool QgsPostgresProjectStorage::removeProject( const QString &uri )
+{
+  QgsPostgresProjectUri projectUri = parseUri( uri );
+  if ( !projectUri.valid )
+    return false;
+
+  QgsPostgresConn *conn = QgsPostgresConnPool::instance()->acquireConnection( projectUri.connInfo.connectionInfo( false ) );
+
+  bool removed = false;
+  if ( _projectsTableExists( *conn, projectUri.schemaName ) )
+  {
+    QString sql( QStringLiteral( "DELETE FROM %1.qgis_projects WHERE name = %2" ).arg( QgsPostgresConn::quotedIdentifier( projectUri.schemaName ), QgsPostgresConn::quotedValue( projectUri.projectName ) ) );
+    QgsPostgresResult res( conn->PQexec( sql ) );
+    removed = res.PQresultStatus() == PGRES_COMMAND_OK;
+  }
+
+  QgsPostgresConnPool::instance()->releaseConnection( conn );
+
+  return removed;
+}
+
+
+QgsPostgresProjectUri QgsPostgresProjectStorage::parseUri( const QString &uri )
+{
+  QUrl u = QUrl::fromEncoded( uri.toUtf8() );
+  QUrlQuery urlQuery( u.query() );
+
+  QgsPostgresProjectUri postUri;
+  postUri.valid = u.isValid();
+
+  QString host = u.host();
+  QString port = u.port() != -1 ? QString::number( u.port() ) : QString();
+  QString username = u.userName();
+  QString password = u.password();
+  QgsDataSourceUri::SslMode sslMode = QgsDataSourceUri::SslPrefer;
+  QString authConfigId;  // TODO: ?
```

Comment:
see blacklisted tests to manage authConfigId to integrate QGIS Auth infrastructure instead to use only basci auth.
btw in this moment it can be tested because we don't have any pki enable docker to test against (I'm slowly working on that when I've some spare time, sorry)

Reply:
hi @luipir this is an older commit, the todo has been addressed already

Reply:
ho sorry, I just saw it in c765295